Travis lint (#2)
EGI-Foundation/datahub-documentation: source/api.rst

.. toctree::

API
---

> It's possible to do all operations using the Onedata API.

Reviewer: ```suggestion
All operations can be performed using the Onedata API
```

Author: Missing ending period, could you add it?

Author: Implemented

The official documentation is: https://onedata.org/#/home/api.

> .. important:: In order to be able to access the `Onedata APIs`, it's required

Reviewer: ```suggestion
.. important:: In order to be able to access the `Onedata APIs`, ian access token is required. See below for instructions on how to generate one.
```

>     to generate an access token.

Reviewer: ```suggestion
```

Getting an API access token
^^^^^^^^^^^^^^^^^^^^^^^^^^^

Tokens have to be generated from the `DataHub` (Onezone) interface as
documented in :ref:`auth-token-using-web-interface` or using a command line
call as documented hereafter.

> The following variables can be exported:

Reviewer: ```suggestion
The following environment variables are required (and can be exported):
```

> 

Reviewer: It's good that the explanation of these variables is given. This makes it comprehensible to the reader. However, I suggest a table to make even clearer, and an example, so that the description is explicit.

Variable Name | Description | Example
-------------------|----------------|------------

* `OIDC_TOKEN`: OpenID Connect Access token, see
  https://wiki.egi.eu/wiki/Federated_Cloud_OpenStack_Providers#Obtaining_an_access_token
  for obtaining it.
* `ONEZONE_HOST`: name or IP of the Onezone host (to use Onezone API).
* `ONEPROVIDER_HOST`: name or IP of the Oneprovider host (to use Oneprovider API).

.. code-block:: console

   export ONEZONE_HOST=datahub.egi.eu
   export OIDC_TOKEN=<OIDC_ACCESS_TOKEN>
   curl -H "X-Auth-Token: egi:$OIDC_TOKEN" -X POST \
     -H 'Content-type: application/json' -d '{}' \

>      https://$ONEZONE_HOST/api/v3/onezone/user/client_tokens

Reviewer: shellcheck will probably complain about bare variables like $ONEZONE_HOST - better to put them between quotes. See [SC2086](https://github.com/koalaman/shellcheck/wiki/Sc2086).

Registering a Handle for a file
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

Prerequisites: access to a Handle service registered in the Onezone.
See
https://onedata.org/#/home/documentation/doc/using_onedata/handle_services.html
for documentation on registering a new Handle service or ask a Onezone
administrator to authorize you to use an existing Handle service already
registered in the Onezone.

> The following variables should be exported:

Reviewer: exporting variables and setting them are slightly different things. Best to be explicit here by changing _exported_ to _set_.

Author: Done.

> * `API_ACCESS_TOKEN`: Onedata API access token

Reviewer: If at all possible, cross-reference Onedata API access token to it's definition somewhere.

Author: Done.
